packages/web3-shared/src/hooks/useTokenDetailed.ts

```diff
@@ -34,3 +44,34 @@ export function useTokenDetailed<P extends EthereumTokenType, Q extends Ethereum
             unreachable(type_)
     }
 }
+
+export function useFungibleTokensDetailed(listOfToken: Pick<FungibleToken, 'address' | 'type'>[]) {
+    const nativeTokenOrders: number[] = []
+    const listOfNativeToken = listOfToken.filter((t, i) => {
+        if (t.type === EthereumTokenType.Native) {
+            nativeTokenOrders.push(i)
+            return true
+        }
+        return false
+    })
+    const listOfERC20Token = listOfToken.filter((t) => t.type === EthereumTokenType.ERC20)
+
+    const { value: nativeTokensDetailed = [], ...asyncNativeResult } = useNativeTokensDetailed(
+        listOfNativeToken as Pick<NativeToken, 'type' | 'address'>[],
+    )
+    const { value: erc20TokensDetailed = [], ...asyncErc20Result } = useERC20TokensDetailed(
+        listOfERC20Token as Pick<ERC20Token, 'type' | 'address'>[],
+    )
+
+    const asyncList = [asyncNativeResult, asyncErc20Result]
+
+    const tokensDetailed = erc20TokensDetailed
+
+    nativeTokenOrders.forEach((order, i) => tokensDetailed.splice(order, 0, nativeTokensDetailed[i]))
```

Comment:
It looks like this variable can be obliterated.

Reply:
removed